docs/proposals/scheduler/agent.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto3";
+
+message Empty {}
+
+
+// ------------------
+// Lifecycle
+// ------------------
+
+message ConnectionRequest {
+    // agent certificate here
+}
+
+enum ConnectionStatus {
+    GRANTED = 0;
+    DENIED = 1;
+}
+
+message ConnectionResponse {
+    ConnectionStatus status = 1;
+}
+
+service LifecycleService {
+    rpc Connect(ConnectionRequest) returns (ConnectionResponse);
+    rpc Disconnect() returns ();
+}
+
+
+// ------------------
+// Status updates
+// ------------------
+
+message NodeStatusUpdateNotification {
+    // update about the node status
+    // such as amount of RAM, CPU, etc.
+}
+
+message InstanceStatusUpdateNotification {
+    // update about instances hosted on the node
+    // such as number of instances down, to reschedule, etc.
```

Comment:
This also must include workload status as well.

Reply:
Please see [this comment](https://github.com/dev-sys-do/orka/pull/2#discussion_r1271311926).